modules/social_features/social_event/src/Form/EnrollActionForm.php

```diff
@@ -86,124 +74,99 @@ class EnrollActionForm extends FormBase implements ContainerInjectionInterface {
+    $instance->configFactory = $container->get('config.factory');
+    $instance->moduleHandler = $container->get('module_handler');
+    $instance->eventEnrollService = $container->get('social_event.enroll');
+    $instance->formBuilder = $container->get('form_builder');
+    return $instance;
   }
 
   /**
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $nid = $this->routeMatch->getRawParameter('node');
-    $current_user = $this->currentUser;
-    $uid = $current_user->id();
-
-    // We check if the node is placed in a Group I am a member of. If not,
-    // we are not going to build anything.
-    if (!is_null($nid)) {
-      if (!is_object($nid)) {
-        $node = $this->entityTypeManager
-          ->getStorage('node')
-          ->load($nid);
-      }
-
-      $groups = $this->getGroups($node);
-
-      // If the user is invited to an event
-      // it shouldn't care about group permissions.
-      $conditions = [
-        'field_account' => $current_user->id(),
-        'field_event' => $node->id(),
-      ];
-
-      $enrollments = $this->entityStorage->loadByProperties($conditions);
-
-      // Check if groups are not empty, or that the outsiders are able to join.
-      if (!empty($groups) && $node->field_event_enroll_outside_group->value !== '1'
-        && empty($enrollments)
-        && social_event_manager_or_organizer() === FALSE) {
+    $current_user = $this->currentUser();
 
-        $group_type_ids = $this->configFactory->getEditable('social_event.settings')
-          ->get('enroll');
+    /** @var \Drupal\node\NodeInterface|string $node */
+    $node = $this->routeMatch->getParameter('node');
 
-        foreach ($groups as $group) {
-          $group_type_id = $group->bundle();
+    if (!($node instanceof NodeInterface)) {
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = $this->entityTypeManager->getStorage('node')->load($node);
+    }
 
-          // The join group permission has never really been used since
-          // this commit. This now means that events in a closed group cannot
-          // be joined by outsiders, which makes sense, since they also
-          // couldn't see these events in the first place.
-          if (in_array($group_type_id, $group_type_ids) && $group->hasPermission('join group', $current_user)) {
-            break;
-          }
+    // We check if the node is placed in a Group I am a member of. If not,
+    // we are not going to build anything.
+    $groups = $this->getGroups($node);
+
+    // If the user is invited to an event
+    // it shouldn't care about group permissions.
+    $enrollments = $this->enrollmentStorage->loadByProperties([
+      'field_account' => $current_user->id(),
+      'field_event' => $node->id(),
+    ]);
+
+    // Check if groups are not empty, or that the outsiders are able to join.
+    if (
+      !empty($groups) &&
+      $node->field_event_enroll_outside_group->value !== '1'
```

Comment:
Should it be TRU instead of '1'?

Reply:
No, because boolean fields return `1` or `0` in Drupal